sherpa-ncnn/csrc/math.h

```diff
@@ -106,10 +106,10 @@ std::vector<int32_t> TopkIndex(const T *vec, int32_t size, int32_t topk) {
   std::vector<int32_t> vec_index(size);
   std::iota(vec_index.begin(), vec_index.end(), 0);
 
-  std::sort(vec_index.begin(), vec_index.end(),
-            [vec](int32_t index_1, int32_t index_2) {
-              return vec[index_1] > vec[index_2];
-            });
+  std::partial_sort(vec_index.begin(), vec_index.begin() + topk,
+                    vec_index.end(), [vec](int32_t index_1, int32_t index_2) {
```

Comment:
```suggestion
                    vec_index.end(), [&vec](int32_t index_1, int32_t index_2) {
```

It was a mistake by not using `&vec`.

Reply:
> there are some test code, vec and &vec not make difference here


Yes, they produce the same result. However, `&vec` is more efficient since it will not copy `vec`.

Hint:
 - `vec`: capture by value -> copy overhead
 - `&vec`: capture by reference -> no copy overhead

Reply:
vec is only a pointer

Reply:
> vec is only a pointer

ah, ok. I though it is a std::vector.